server/cli/src/plugins.rs

```diff
@@ -125,3 +163,70 @@ fn process_manifest(bundle: &mut PluginBundle, path: &PathBuf) -> Result<(), Err
 
     Ok(())
 }
+
+/// username, password, url should come from config, like in reports (in the new show command)
+pub(super) async fn install_plugin_bundle(
+    InstallPluginBundle {
+        path,
+        url,
+        username,
+        password,
+    }: InstallPluginBundle,
+) -> Result<(), Error> {
+    let api = Api::new_with_token(url.clone(), username, password).await?;
+
+    let uploaded_file = api.upload_file(path).await?;
+
+    let upload_result = api
+        .gql(
+            INSTALL_PLUGINS,
+            json!( { "fileId": uploaded_file.file_id} ),
+            Some("CentralServerMutationNode"),
+        )
+        .await?;
+
+    println!(
+        "Result:{}",
+        serde_json::to_string_pretty(&upload_result).unwrap()
+    );
+
+    Ok(())
+}
+
+/// username and password should come from config, like in reports (and url too)
+pub(super) async fn generate_and_install_plugin_bundle(
+    GenerateAndInstallPluginBundle {
+        in_dir,
+        url,
+        username,
+        password,
+    }: GenerateAndInstallPluginBundle,
+) -> Result<(), Error> {
+    let out_file = PathBuf::from("report_temp.json");
+
+    generate_plugin_bundle(GeneratePluginBundle {
+        in_dir,
+        out_file: out_file.clone(),
+    })?;
+
+    let api = Api::new_with_token(url.clone(), username, password).await?;
```

Comment:
Potentially can call the function `install_plugin_bundle` from with in this one, but I agree it's not too much code duplication, so I'm happy with this..

Reply:
Agree, thanks: 532f988b3270fd2a061c470fa42ece2b8fee261b